### PR TITLE
drop CRD spec.scope default value in doc

### LINKED ISSUE #84257: CustomResourceDefinitionSpec missing default scope value
**What happened**:
The comments for the `Scope` field in `apiextensionsv1.CustomResourceDefinitionSpec` state that the default value is `Namespaced`, however no default value is set.

- Default value and comment removed: https://github.com/kubernetes/kubernetes/commit/05d8b448d274e30772750fcea71b10c8177e556e
- Comment re-added: https://github.com/kubernetes/kubernetes/commit/e60b82a7b83b982a28399fb744b05da0450b6346

**What you expected to happen**:

Maximally, I expect a valid default value to be set for non-pointer "enum" types.

Minimally, I expect the comments and behavior to match.

**How to reproduce it (as minimally and precisely as possible)**:
N/A

**Anything else we need to know?**:
N/A

**Environment**:
N/A

### PR DESCRIPTION
**Pull request by danielqsj**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #84257

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/generated.proto [hunk 1]
```diff
@@ -224,7 +224,7 @@ message CustomResourceDefinitionSpec {
   optional CustomResourceDefinitionNames names = 3;
 
   // scope indicates whether the defined custom resource is cluster- or namespace-scoped.
-  // Allowed values are `Cluster` and `Namespaced`. Default is `Namespaced`.
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/generated.proto]: revert this change, this default is accurate in v1beta1

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ type CustomResourceDefinitionSpec struct {
 	// names specify the resource and kind names for the custom resource.
 	Names CustomResourceDefinitionNames `json:"names" protobuf:"bytes,3,opt,name=names"`
 	// scope indicates whether the defined custom resource is cluster- or namespace-scoped.
-	// Allowed values are `Cluster` and `Namespaced`. Default is `Namespaced`.
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go]: revert here as well

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/generated.proto (hunk 1 above)
Comment (danielqsj) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/generated.proto]: done, thanks :)

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go (hunk 2 above)
Comment (danielqsj) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go]: done, thanks :)
